tsdb/chunks/chunks.go

```diff
@@ -311,6 +314,50 @@ func MergeChunks(a, b chunkenc.Chunk) (*chunkenc.XORChunk, error) {
 	return newChunk, nil
 }
 
+// Avoid chunks with more than a certain number of samples
+// Max of 120 samples per chunk results in best compression ratio.
+func BreakLargeChunks(chks []Meta, limit int) ([]Meta, error) {
+	newChunksArray := make([]Meta, 0 , len(chks)*2)
```

Comment:
I would go a step further and ask whether preallocting is even worth it. What's the expected average case for this func (ie: does it result in breaking chunks more often than not?). What's the expected number of chunks to normally be passed in here? If it's low it seems likely that relying on append to allocate more space is performant enough.

It would be ideal to benchmark these different behaviours and cases, but tbh if the expected chunk count is low (in the 10s or less, which seems plausible but I don't profess to have a deep understanding here) I wouldn't bother and would just let `append` handle all allocations.

Reply:
Yeah I was a bit concerned about `len(chks)*2 `. I set it randomly but now I agree that it would have been better to do benchmarking. For now I'll just remove the preallocation.